armi/reactor/tests/test_reactors.py

```diff
@@ -252,6 +253,25 @@ def setUp(self):
             self.directoryChanger.destination, customSettings={"trackAssems": True}
         )
 
+    def test_factorySortSetting(self):
+        # get a sorted Reactor (the default)
+        cs = settings.Settings(fName="armiRun.yaml")
+        r0 = reactors.loadFromCs(cs)
+
+        # get an unsorted Reactor (for whatever reason)
+        customSettings = {CONF_SORT_REACTOR: False}
+        cs = cs.modified(newSettings=customSettings)
+        r1 = reactors.loadFromCs(cs)
+
+        # the reactor / core should be the same size
+        self.assertEqual(len(r0), len(r1))
+        self.assertEqual(len(r0.core), len(r1.core))
+
+        # the reactor / core should be in a different order
+        a0 = [a.name for a in r0.core]
+        a1 = [a.name for a in r1.core]
+        self.assertNotEqual(a0, a1)
```

Comment:
Thoughts on including this assertion in the other test over in test_database3.py as well?

Reply:
It wouldn't work the same. The assemblies in THIS test file are added to the reactor in order.  Thus, I would have to change the tests here to have them create a Reactor in a non-sorted order by default.  But there are plugins that import these tests, and I'm _really_ trying to keep this PR from requiring PRs in downstream plugins. (Especially for this "Setting" bonus feature that I'm just going to want to rip our in a few months.)

Reply:
>  I'm really trying to keep this PR from requiring PRs in downstream plugins.

Great call.